src/js/graphCanvas.js

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2016 Michael Warren Skirpan
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+*/
+
+(function (d3, $, fluid) {
+
+    "use strict";
+
+    fluid.defaults("floe.dataDashboard.graphCanvas", {
+        gradeNames: ["fluid.viewComponent"],
+        selectors: {
+            dataFields: ".floec-graphCanvas-fields",
+            graphRepresentation: ".floec-graphCanvas-representation",
+            dataOptions: ".floec-graphCanvas-options",
+            buttons: ".floec-graphCanvas-buttons"
+        },
+        model: {
+            graphType: "{floe.dataDashboard.graphSelector}.model.graph",
+            workingData: "{floe.dataDashboard}.model.rawData",
+            currentSelection: {
+            },
+            selectionState: null,
+            
+        },
+        components: {
+            dataFields: {
+                type: "floe.dataDashboard.graphCanvas.dataFields",
+                container: "{graphCanvas}.dom.dataFields",
+                createOnEvent: "floe.dataDashboard.graphCanvas.dataParsed"
+            },
+            graphRepresentation: {
+                type: "floe.dataDashboard.graphCanvas.graphRepresentation",
+                container: "{graphCanvas}.dom.graphRepresentation"
+            },
+            dataOptions: {
+                type: "floe.dataDashboard.graphCanvas.dataOptions",
+                container: "{graphCanvas}.dom.dataOptions"
+            },
+        },
+        events: {
+            dataParsed: null
+
+        },
+        listeners: {
+            //Only while testing
+            "onCreate.testBuilder" : {
+                funcName: "floe.dataDashboard.graphCanvas.graphRepresentation.buildTestLine",
+                args: ["{that}"]
+            },
+            "onCreate.parseData" : {
+                funcName: "{graphCanvas}.parseTheData",
+                args: "[{that}]"
+            }
+            
+        }
+
+    });
+
+    floe.dataDashbaord.graphCanvas.parseTheData = function (that) {
```

Comment:
Cool, changed this. Sorry I've been in the process of setting up the fields for different kinds of click interactions. I pushed up things before getting to a test. Any other relics will be gone in the next push I make.